File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/viewmodel/item/ItemTabVModel.java
package io.ganguo.demo.viewmodel.item;

import android.support.annotation.DrawableRes;
import android.support.annotation.StringDef;
import android.view.View;

import io.ganguo.demo.R;
import io.ganguo.demo.databinding.ItemTabIconViewBinding;
import io.ganguo.vmodel.BaseViewModel;
import io.ganguo.library.ui.view.ActivityInterface;

/**
 * Created by leo on 16/7/11.
 * item tab
 */
public class ItemTabVModel extends BaseViewModel<ActivityInterface<ItemTabIconViewBinding>> {
    @DrawableRes
    private int icon;
    private String title;

    public ItemTabVModel(int icon, String title) {
        this.icon = icon;
        this.title = title;
    }

    public int getIcon() {
        return icon;
    }

    public void setIcon(int icon) {
        this.icon = icon;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    @Override
    public int getItemLayoutId() {
        return R.layout.item_tab_icon_view;
    }

    @Override
    public void onViewAttached(View view) {

    }
}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/viewmodel/item/ItemTabVModel.kt
package io.ganguo.demo.viewmodel.item

import androidx.annotation.DrawableRes
import android.view.View

import io.ganguo.demo.R
import io.ganguo.demo.databinding.ItemTabIconViewBinding
import io.ganguo.viewmodel.core.BaseViewModel
import io.ganguo.viewmodel.core.viewinterface.ActivityInterface

/**
 * Created by leo on 16/7/11.
 * item tab
 */
class ItemTabVModel(@field:DrawableRes
                    var icon: Int, var title: String?) : BaseViewModel<ActivityInterface<ItemTabIconViewBinding>>() {

    override fun onViewAttached(view: View) {
    }

    override val layoutId: Int by lazy { R.layout.item_tab_icon_view }
}
